include/deal.II/lac/trilinos_vector.h

```diff
@@ -718,7 +718,14 @@ namespace TrilinosWrappers
                                             )));
         }
 
-      reinit (vector_partitioner(), v);
+      const Epetra_Map &map = vector_partitioner();
+      const int size = map.NumMyElements();
+
+      // Need to copy out values, since the deal.II might not use doubles, so
+      // that a direct access is not possible.
+      for (int i=0; i<size; ++i)
+        (*vector)[0][i] = v(gid(map,i));
```

Comment:
`gid` is one of our own [function](https://github.com/Rombur/dealii/blob/ef52c69409b6ee3e7d0233e7937172db9c5d6a7f/include/deal.II/lac/trilinos_vector.h#L51). The function in trilinos is either `map.GID` or `map.GID64` if using 64 bits.